set_transformers/tensorflow/layers.py

```diff
@@ -0,0 +1,456 @@
+from typing import List, Dict, Tuple
+
+import tensorflow as tf
+
+from tensorflow.keras.layers import (
+    MultiHeadAttention,
+    Dense,
+    LayerNormalization
+)
+
+
+class RFF(tf.keras.layers.Layer):
+    def __init__(
+        self,
+        units_per_layers: List[int],
+        dense_kwargs: Dict = dict(activation="relu"),
+        **kwargs
+    ):
+        """Row-wise Feed-Forward (rFF) layer.
+        Processes each row independently using an mlp
+        y = MLP(x)
+
+        Parameters
+        ----------
+        units_per_layers : List[int]
+            units per Dense layer
+        dense_kwargs : Dict, optional
+            aditional kwargs for Dense layers,
+            by default {"activation": "relu"}
+        """
+        super().__init__(**kwargs)
+
+        self.mlp = tf.keras.Sequential(
+            layers=[
+                Dense(
+                    units=units,
+                    **dense_kwargs
+                )
+                for units in units_per_layers
+            ]
+        )
+
+    def call(
+        self,
+        x: tf.Tensor
+    ) -> tf.Tensor:
+        """y = MLP(x)
+
+        Parameters
+        ----------
+        x : tf.Tensor
+            shape: (batch_size, n, d)
+
+        Returns
+        -------
+        tf.Tensor
+            shape: (batch_size, n, d)
+        """
+        return self.mlp(x)
```

Comment:
Why call the tf subclassing API when this is just a MLP layer? Why not a function returning the model? Using the subclassing API implies some restrictions due to its nature (vs the sequential or functional api)

Reply:
What I'm refering to: https://github.com/tensorflow/tensorflow/issues/30955#issuecomment-520291616